testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/BaseTestAvroSchemaEvolution.java

```diff
@@ -0,0 +1,283 @@
+    public static final String REMOVED_COLUMN_SCHEMA = "file:///docker/presto-product-tests/avro/remove_column_schema.avsc";
+    public static final String ADDED_COLUMN_SCHEMA = "file:///docker/presto-product-tests/avro/add_column_schema.avsc";
+    public static final String CHANGE_COLUMN_TYPE_SCHEMA = "file:///docker/presto-product-tests/avro/change_column_type_schema.avsc";
+    public static final String INCOMPATIBLE_TYPE_SCHEMA = "file:///docker/presto-product-tests/avro/incompatible_type_schema.avsc";
+
+    private final String tableName;
+    private final String columnsInTableStatement;
+    private final String selectStarStatement;
+    private final List<Partition> partitions;
+
+    protected BaseTestAvroSchemaEvolution(String tableName, Partition... partitions)
+    {
+        this.tableName = tableName;
+        this.columnsInTableStatement = "SHOW COLUMNS IN " + tableName;
+        this.selectStarStatement = "SELECT * FROM " + tableName;
+        this.partitions = Arrays.asList(partitions);
+    }
+
+    @BeforeTestWithContext
+    public void createTestTable()
+    {
+        query(format(
+                "CREATE TABLE %s (" +
+                        "  dummy_col VARCHAR" +
+                        (partitions.isEmpty() ? "" : ", " + getPartitionsAsString(", ", p -> p.getColumnDeclaration())) +
+                        ")" +
+                        "WITH (" +
+                        "  format='AVRO', " +
+                        "  avro_schema_url='%s'" +
+                        (partitions.isEmpty() ? "" : ", partitioned_by=ARRAY['" + getPartitionsAsString("', '", p -> p.getName()) + "']") +
+                        ")",
+                tableName,
+                ORIGINAL_SCHEMA));
+        insertData(tableName, 0, "'stringA0'", "0");
+    }
+
+    @AfterTestWithContext
+    public void dropTestTable()
+    {
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+    }
+
+    @Test(groups = AVRO)
+    public void testSelectTable()
+    {
+        assertThat(query(format("SELECT string_col FROM %s", tableName)))
+                .containsExactlyInOrder(row("stringA0"));
+    }
+
+    @Test(groups = AVRO)
+    public void testCreateTableLike()
+    {
+        String createTableLikeName = "test_avro_like";
+        query(format(
+                "CREATE TABLE %s (LIKE %s INCLUDING PROPERTIES)",
+                createTableLikeName,
+                tableName));
+
+        insertData(createTableLikeName, 0, "'stringA0'", "0");
+
+        assertThat(query(format("SELECT string_col FROM %s", createTableLikeName)))
+                .containsExactlyInOrder(row("stringA0"));
+        query("DROP TABLE IF EXISTS " + createTableLikeName);
+    }
+
+    @Test(groups = AVRO)
+    public void testInsertAfterSchemaEvolution()
+    {
+        assertThat(query(selectStarStatement))
+                .containsOnly(
+                        createRow(0, "stringA0", 0));
```

Comment:
Since we have one row to assert, can we inline it ?

Reply:
I wanted to keep it aligned with all assertions, as it's easier to navigate over the code.
All createRow are indented in same way

Reply:
If you think this is obsolete let me know I will inline them.